# MAL Analysis Project 
Created by TomPancakes

Importing Libraries 

In [1]:
import pandas as pd
import matplotlib as plt



Import main file, and create pandas data frame using api data

In [2]:
#import and run main.py (makes api request)
import os

if os.path.exists("anime.json"):
    df = pd.read_csv('anime.json')
else: 
    import main #Run main python script only if no csv found
    df = pd.DataFrame(main.fetched_data)
    df.to_csv('anime.json', index=False)

df[['title', 'averageScore', 'popularity']].head(10)



Attempting to grab api data
Fetched page 1/20
Fetched page 2/20
Fetched page 3/20
Fetched page 4/20
Fetched page 5/20
Fetched page 6/20
Fetched page 7/20
Fetched page 8/20
Fetched page 9/20
Fetched page 10/20
Fetched page 11/20
Fetched page 12/20
Fetched page 13/20
Fetched page 14/20
Fetched page 15/20
Fetched page 16/20
Fetched page 17/20
Fetched page 18/20
Fetched page 19/20
Fetched page 20/20


,title,averageScore,popularity
0,Attack on Titan,85,977679
1,Demon Slayer: Kimetsu no Yaiba,83,925923
2,Death Note,84,891203
3,JUJUTSU KAISEN,84,890467
4,My Hero Academia,77,817876
5,Hunter x Hunter (2011),89,783448
6,One-Punch Man,83,722475
7,Tokyo Ghoul,76,691868
8,Attack on Titan Season 2,85,686143
9,ONE PIECE,87,682334


Selection Cheatsheet:

In [3]:
#Selection Practice: 
#Select by column
avg_rating = df["averageScore"].mean() #Select by avgScore column (make into series), use pandas .mean() method
print(f"The average rating for all anime is: {avg_rating}\n" )

#Select by row
print(df.loc[0], "\n") #Grab by index. (I.e., most popular show)
print(df.loc[df['title'] == "Frieren: Beyond Journey’s End"]) 



The average rating for all anime is: 75.387

id                                          16498
title                             Attack on Titan
averageScore                                   85
episodes                                     25.0
season                                     SPRING
seasonYear                                 2013.0
status                                   FINISHED
format                                         TV
popularity                                 977679
genres          [Action, Drama, Fantasy, Mystery]
Name: 0, dtype: object 

        id                          title  averageScore  episodes season  \
67  154587  Frieren: Beyond Journey’s End            91      28.0   FALL   

    seasonYear    status format  popularity                       genres  
67      2023.0  FINISHED     TV      410864  [Adventure, Drama, Fantasy]  


The average score for the 1000 most popular anime on AniList is 75.39. 

In [15]:
#average score by genre. 

# Dataset currently stores genres as a list of multiple genres.
# Thus, to effectively compare individual genres, a new data frame will be created
# DF_exploded contains individual entries for EACH genre of EACH show

df_exploded = df.explode('genres')
genre_scores = df_exploded.groupby('genres')['averageScore'].mean()
print(genre_scores.sort_values(ascending=False))

#genre_scores.plot(kind= 'bar')





genres
Music            79.590909
Sports           79.200000
Drama            78.121359
Psychological    78.105263
Thriller         77.954545
Mystery          77.093168
Mahou Shoujo     76.500000
Slice of Life    76.146617
Supernatural     76.090253
Sci-Fi           75.837349
Adventure        75.722022
Mecha            75.322581
Action           75.010267
Comedy           74.475584
Fantasy          74.349727
Romance          73.241758
Horror           72.928571
Ecchi            68.953846
Name: averageScore, dtype: float64


### Interpretation
The table above shows some interesting results.

Firstly, it seems that higher rated genres tend to be more uncommon, where the lower rated genres are generally more common. For example, music, sports, drama are in the top 3, where action, comedy, fantasy, romance and horror are in the bottom segment. An interpretation of this could be that works in already saturated genres are seen as more generic and less standout. Whereabouts works which are in less crowded genres are seen as more unique and interesting.

The genre of ecchi being the lowest rated is expected, as series with heavy fan service tend to be heavily criticised. Horror being also placed low is also interesting. Perhaps this suggests horror generally is difficult to translate into an animated medium like anime. 

In [16]:
#random.

print(df['genres'][14]) #get genres of 14th entry. 



['Drama', 'Romance', 'Slice of Life']
